### PR TITLE
Replace MaterialUI button with our implementation

### PR DESCRIPTION
**Pull request by ajronn**

### CODE REVIEW

File: src/ui/button/Button.scss [hunk 1]
```diff
@@ -42,19 +54,17 @@
...
-
 .iconButton {
   width: 50px;
   height: 50px;
+  display: inline-flex;
 }
+
+.textIcon{
+  display: flex;
+  align-items: center;
```
Comment (Czaroos) [inline src/ui/button/Button.scss:65]: instead lines 64 and 65 use one line `@include row`

**Review by Czaroos — CHANGES_REQUESTED** [review]

File: src/ui/button/Button.scss (hunk 1 above)
Comment (ajronn) [reply src/ui/button/Button.scss:65]: This style is temporary cause we need to merge all branches which use Button component for correct styling

File: src/modules/main/navbar/user-section/UserSection.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ const UserSection = () => {
       <Guard.Unprotected>
         <>
           <NavLink to="/login">
-            <Button className={csx.logInBtn}>LOG IN</Button>
+            <Button style={csx.logInBtn}>LOG IN</Button>
```
Comment (polubis) [inline src/modules/main/navbar/user-section/UserSection.tsx:41]: style ???

File: src/ui/button/Button.scss [hunk 3]
```diff
@@ -1,7 +1,19 @@
 @import 'styles/mixins';
 
-.button,
-.iconButton {
+button {
+  box-sizing: border-box;
```
Comment (polubis) [inline src/ui/button/Button.scss:4]: box-sizing border-box is not needd

File: src/ui/button/Button.scss [hunk 4]
```diff
@@ -1,7 +1,19 @@
...
+  border: none;
+  outline: none;
+  background-color: transparent;
+  position: relative;
+  padding: 9px 20px !important;
+  cursor: pointer; 
+  font-size: 14px;
+  border-radius: 4px;
+  font-family: $fontPrimary;
+  font-size: 14px;
```
Comment (polubis) [inline src/ui/button/Button.scss:14]: double font-size

File: src/ui/button/Button.scss [hunk 5]
```diff
@@ -1,7 +1,19 @@
 @import 'styles/mixins';
 
-.button,
-.iconButton {
+button {
+  box-sizing: border-box;
+  border: none;
+  outline: none;
+  background-color: transparent;
```
Comment (polubis) [inline src/ui/button/Button.scss:7]: background instead of bc

File: src/ui/button/Button.scss [hunk 6]
```diff
@@ -1,7 +1,19 @@
...
+  position: relative;
+  padding: 9px 20px !important;
```
Comment (polubis) [inline src/ui/button/Button.scss:9]: Why you used important if this is module without any parent stylings

File: src/ui/button/Button.tsx [hunk 7]
```diff
@@ -4,37 +4,49 @@ import { Button as MuiButton, IconButton as MuiIconButton } from '@material-ui/c
 
 import csx from './Button.scss';
 
+const Btn = (props, children) => {
```
Comment (polubis) [inline src/ui/button/Button.tsx:7]: ???

File: src/ui/button/Button.tsx [hunk 8]
```diff
@@ -4,37 +4,49 @@ import { Button as MuiButton, IconButton as MuiIconButton } from '@material-ui/c
...
-      React.ButtonHTMLAttributes<HTMLButtonElement>,
-      HTMLButtonElement
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
     > {
     children: React.ReactNode;
-    variant?: 'default' | 'icon';
+    variant?: 'default' | 'icon' | 'texticon';
     theme?: 'primaryDark' | 'primaryTransparent' | 'danger';
+    style?: Object;
```
Comment (polubis) [inline src/ui/button/Button.tsx:24]: style should be removed

**Review by polubis — CHANGES_REQUESTED** [review]
Summary:
I know your implementation is based on previous. But as I said in recent workshops we need to create our components with easy to use API.

So button component should be simplified. Try to add an option to pass through props custom class from the parent component. className prop and assign this to button classNames. Also, try design button component API first. Check all button elements in the application and think about how many variants/themes we need. 

Try design first approach - use some tools for that - draw.io.

So design button component - maybe create a ButtonBase component with a redundant interface and create several variants like MaterialUI or maybe use other approaches. The goal is to do it in a scalable and easy to use way - much easier than previous approach.

File: src/modules/main/navbar/user-section/UserSection.tsx (hunk 2 above)
Comment (ajronn) [reply src/modules/main/navbar/user-section/UserSection.tsx:41]: style was repleced with className

File: src/ui/button/Button.scss (hunk 4 above)
Comment (ajronn) [reply src/ui/button/Button.scss:14]: was fixed

File: src/ui/button/Button.scss (hunk 5 above)
Comment (ajronn) [reply src/ui/button/Button.scss:7]: was fixed

File: src/ui/button/Button.scss (hunk 6 above)
Comment (ajronn) [reply src/ui/button/Button.scss:9]: was fixed

File: src/ui/button/Button.tsx (hunk 8 above)
Comment (ajronn) [reply src/ui/button/Button.tsx:24]: was removed

File: src/ui/button/Button.tsx (hunk 7 above)
Comment (ajronn) [reply src/ui/button/Button.tsx:7]: will be removed along with the removal of the MUI dependent component

**Review by polubis — APPROVED** [review]
This task will be merged later in new feature branch - review done all ok

**Pull request closed without merging**